### DISCUSSION TITLE
How to retrieve info on user-disks: ids and names

### DISCUSSION THREAD

**Discussion by ymelnikova** [biigle/core#844] [Q&A]
Hello community, 
I couldn't find API command to retrieve pairs of ids and corresponding names of the user disks.
What would be a workaround?

**Comment by mzur** [comment]
Hi, as you already noticed, there is no API endpoint to list user disks. This is because it isn't needed by the BIIGLE UI so far. However, I'd gladly accept a pull request that implements the API endpoint. Here are the BIIGLE [developing instructions](https://github.com/biigle/core/blob/master/DEVELOPING.md). The new API endpoint could be added as a new `index()` method in the [`UserDiskController`](https://github.com/biigle/user-disks/blob/main/src/Http/Controllers/Api/UserDiskController.php). User disks can be retrieved like in the [view controller](https://github.com/biigle/user-disks/blob/42a2ffbb10c6864cb349be81b3427efe9f0ebc5b/src/Http/Controllers/Views/UserDiskController.php#L22). The new API endpoint also needs a [test](https://github.com/biigle/user-disks/blob/main/tests/Http/Controllers/Api/UserDiskControllerTest.php).

**Reply by ymelnikova** [reply]
Hi Martin, 
Thank you for your reply! As I am not a PHP developer, it would be a bit difficult to provide the pull-request.  Maybe at a later point.
For now, as a workaround, I retrieve the id of the disk at the moment of creation.  

For context, I want to create a new volume in the project,  and I need to provide the media url for that .  For example, disk-01://volumes/1, but then I need to know the id of my storage disk.  It would be convenient to retrieve it by the name of the disk. But the workaround I described above works as well. 

Yulia.

**Reply by mzur** [reply]
Ok, good that it works for you. I'd be interested in your use case. Do you have so many storage disks that you create them via the API? Otherwise, if you have only 1-2 disks, you can also find out their ID from the URL of the "edit storage disk" view.
